### PR TITLE
Update unmaintained packages

### PR DESCRIPTION
**Pull request by matthiasbeyer**
###### Motivation for this change

This PR updates a bunch of _unmaintained_ packages (as reported by monitor). I test-build all of these PRs (in the order they are committed) using a Debian 8 VM with nix installed. Details below.

I updated them by fetching the packages and running `nix-script update-package-def` (see [my nixos-scripts](https://github.com/matthiasbeyer/nixos-scripts))with appropriate flags for building etc. in a loop.

Please note that I did not execute them or test them in any way besides the build (`nix-build -A <pkg> --cores 8 -j 16`).
###### Things done
- [ ] Tested using sandboxing
  ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS,
    or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
  - [ ] NixOS
  - [ ] OS X
  - [x] Linux
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

Details on the machine I used to build:

``` bash
m@banane:~$ uname -a
Linux banane 3.16.0-4-amd64 #1 SMP Debian 3.16.7-ckt25-2 (2016-04-08) x86_64 GNU/Linux
m@banane:~$ cat /etc/os-release 
PRETTY_NAME="Debian GNU/Linux 8 (jessie)"
NAME="Debian GNU/Linux"
VERSION_ID="8"
VERSION="8 (jessie)"
ID=debian
HOME_URL="http://www.debian.org/"
SUPPORT_URL="http://www.debian.org/support"
BUG_REPORT_URL="https://bugs.debian.org/"
m@banane:~$ nix-env --version
nix-env (Nix) 1.11.2
m@banane:~$ 
```

### CODE REVIEW

**Comment by groxxda** [comment]
:-1: on the wayland update. `.91` is alpha, there is already `.93` which is RC1. But I think our policy is not to update to unstable versions if there is no reason.
If this gets merged please also update weston to the same version.

**Comment by bjornfor** [comment]
This is not a full PR review, but I noticed that you have a "evince: 3.18.2 -> 3.20.0" in there. I'm pretty sure that update is planned to be done in the (separate) GNOME 3.20 PR: https://github.com/NixOS/nixpkgs/pull/14366.

**Comment by groxxda** [comment]
The postfix update isn't the newest version (`3.1.1`)
There's a imcompat change according to their [3.1 releasenotes](http://cdn.postfix.johnriley.me/mirrors/postfix-release/official/postfix-3.1.1.RELEASE_NOTES) :

>[Incompat 20150719] The default Diffie-Hellman non-export prime was updated from 1024 to 2048 bits, because SMTP clients are starting to reject TLS handshakes with primes smaller than 2048 bits.

So this possibly needs review?

**Comment by matthiasbeyer** [comment]
@groxxda Will remove the patch for wayland. Not sure on the postfix part, shall I remove this one, too?

---

@bjornfor I will remove the patch for evince.

**Comment by groxxda** [comment]
I my opinion the postfix update should be fine - but I'd rather update to the newest patchset (3.1.1) :wink: 
By the way: You might want to check your `git.name` setting.

**Comment by mbakke** [comment]
I just made another PR for feh: #15859

File: pkgs/servers/asterisk/default.nix [hunk 1]
```diff
@@ -2,11 +2,11 @@
 
 stdenv.mkDerivation rec {
   name = "asterisk-${version}";
-  version = "13.6.0";
+  version = "13.9.1";
 
   src = fetchurl {
     url = "http://downloads.asterisk.org/pub/telephony/asterisk/asterisk-${version}.tar.gz";
```
Comment (groxxda) [inline pkgs/servers/asterisk/default.nix]: Can you change to `http://downloads.asterisk.org/pub/telephony/asterisk/releases/${name}.tar.gz` (Note the `releases/`, so the link remains working when they release a new version)?

**Comment by matthiasbeyer** [comment]
@groxxda What is about my `git.name` setting?

@mbakke Should I remove my `feh` patch?

**Comment by joachifm** [comment]
Can't say I'm such a huge fan of blindly updating important packages like postfix and postgres ...

**Comment by groxxda** [comment]
@matthiasbeyer Github says:
> matthiasbeyer committed with Your Name 19 hours ago

**Comment by matthiasbeyer** [comment]
@groxxda Oh, yes. I will rewrite the history once the discussions in here came to a conclusion anyways, I will fix that then.

@joachifm If you feel uncomfortible with that I will take the patches out if you want to.

**Comment by joachifm** [comment]
@matthiasbeyer if you do not intend to test the packages, I suggest at least factoring them out in separate PRs, to allow their maintainers to provide some feedback, instead of making this an all-or-nothing kind of deal.

**Comment by mbakke** [comment]
> @mbakke Should I remove my feh patch?

Yes, a newer version is already merged.

**Comment by matthiasbeyer** [comment]
@groxxda postfix patch isn't provided by monitor, so I'll remove the patch instead of adding one update by hand.

**Comment by mbakke** [comment]
I've extracted the `less` update into #15868 as I've been meaning to modify the expression for a while.

**Comment by joachifm** [comment]
varnish, redis, musl, ncurses, openvpn, and asterisk also seem like stuff that should be submitted separately. R can be dropped, as its already out of date. The less patch has been merged.

**Comment by matthiasbeyer** [comment]
Removed R and less.

**Comment by joachifm** [comment]
vifm, signing-party, icewm, discount, and diffstat have been merged.

**Comment by matthiasbeyer** [comment]
@joachifm removed the updates you named from the PR.

**Comment by bjornfor** [comment]
I hope we soon get Hydra to build github PRs. I think that will help a lot with PRs such as this. (If nothing breaks, even NixOS tests, we can safely merge.)

**Comment by domenkozar** [comment]
@bjornfor I'll work on that soon.

@matthewbauer did these get merged? In general it's easier to have a PR per minimal mergable branch, most of our big PRs just hang around.

**Comment by matthiasbeyer** [comment]
Funny. I just closed this PR because of its age ...

**Comment by domenkozar** [comment]
@matthewbauer that's why I said that, it's not that we do not want these changes, it's just hard to get them through. 

Once we have a PR testing on Hydra situation will be different, but at the moment our workflow doesn't allow us easy testing :/

**Comment by domenkozar** [comment]
Sorry about that, if you find motivation to split these into smaller blocks I'm sure we'll get those merged eventually.

**Comment by matthiasbeyer** [comment]
@domenkozar I guess you mean me... no, I guess I will wait until we have hydra working on PRs and then start to file PRs like this one. You are right, before we do not have decent testing, PRs like this do not make a lot of sense.

**Comment by vcunat** [comment]
I certainly agree that that big PRs are hard to get through in general. On most days I personally don't have enough time to finish a big one, so I tend to focus on smaller ones.

If these are changes from nixpkgs monitor, that one also attempts to build the changes (besides the submitter in this case). The PR testing on Hydra would do the same, right? ... perhaps including all reverse dependencies, though.

BTW, I think a couple of these packages have been updated independently already, but that shouldn't matter.

**Comment by matthiasbeyer** [comment]
Yeah, I have a script to where I simply paste a monitor patch link and it applies the patch and also test-compiles it.

I really should tweak this script to only apply if compiling works. Then I could use the HPC cluster from my university for something good... :smile:

**Pull request closed without merging**